pkg/provisioner/ironic/provision_test.go

```diff
@@ -745,7 +745,9 @@ func TestGetUpdateOptsForNodeWithRootHints(t *testing.T) {
 			t.Logf("expected: %v", e)
 			var update nodes.UpdateOperation
 			for _, patch := range patches {
-				update = patch.(nodes.UpdateOperation)
+				updateOperation, ok := patch.(nodes.UpdateOperation)
+				require.True(t, ok, "expected patch to be UpdateOperation")
+				update = updateOperation
```

Comment:
nit: We can skip the extra variable throughout this file
```suggestion
				update, ok := patch.(nodes.UpdateOperation)
				require.True(t, ok, "expected patch to be UpdateOperation")
```

Reply:
Just have to wonder what have I been thinking while doing this... :laughing: